addons/website_event/controllers/main.py

```diff
@@ -42,6 +42,28 @@ def _get_events_search_options(self, slug_tags, **post):
             'country': post.get('country'),
         }
 
+    def _get_default_country_search(self, events):
+        """
+            Find the country from geoip/user and return it only if
+            future events exist for that country, else fallback to 'all'
+        """
```

Comment:
I would do a standard format `:param: ...` docstring to specify that `events` is the set of events returned from the search on the main event page :)